src/transformers/modeling_flax_pytorch_utils.py

```diff
@@ -180,16 +191,16 @@ def convert_pytorch_state_dict_to_flax(pt_state_dict, flax_model):
     flax_state_dict = {}
 
     load_model_with_head_into_base_model = (model_prefix not in flax_model_params) and (
-        model_prefix in {k.split(".")[0] for k in pt_state_dict.keys()}
+        model_prefix in {k.split(".")[0] for k in state_dict.keys()}
     )
     load_base_model_into_model_with_head = (model_prefix in flax_model_params) and (
-        model_prefix not in {k.split(".")[0] for k in pt_state_dict.keys()}
+        model_prefix not in {k.split(".")[0] for k in state_dict.keys()}
     )
 
     # Need to change some parameters name to match Flax names
-    for pt_key, pt_tensor in pt_state_dict.items():
+    for pt_key, pt_tensor in state_dict.items():
         pt_tuple_key = tuple(pt_key.split("."))
-        is_bfloat_16 = weight_dtypes[pt_key] == torch.bfloat16
+        is_bfloat_16 = is_bfloat16(pt_state_dict[pt_key])
```

Comment:
And then this drops out to:
```suggestion
        is_bfloat_16 = pt_state_dict[pt_key].dtype == bfloat16
```